mp/src/game/shared/momentum/mom_gamemovement.cpp

```diff
@@ -1283,31 +1283,77 @@ void CMomentumGameMovement::CategorizePosition()
         {
             if (player->GetGroundEntity() == nullptr)
             {
-                Vector rampVelocity = mv->m_vecVelocity;
+                Vector vecNextVelocity = mv->m_vecVelocity;
 
                 // Apply half of gravity as that would be done in the next tick before movement code
-                rampVelocity[2] -= (player->GetGravity() * GetCurrentGravity() * 0.5 * gpGlobals->frametime);
+                vecNextVelocity -= (player->GetGravity() * GetCurrentGravity() * 0.5 * gpGlobals->frametime);
```

Comment:
Did you mean to change this from just affecting the Z velocity to the entire vector?

Reply:
Good catch, only meant the z-component.